util/xivapi.ts

```diff
@@ -94,10 +95,11 @@ export class XivApi {
       let jsonResult;
       try {
         const response = await fetch(url);
-        if (!response.ok)
-          throw new Error(`Network error occurred fetching API results.`);
         jsonResult = (await response.json()) as XivApiResult;
-        const pageNum = jsonResult.Pagination.Page;
+        if (!response.ok)
+          throw new Error(`Error occurred fetching API results.`);
```

Comment:
The error message here changed, was this intentional? And if so, and this is a more generic error, is there any additional information we can glean from `response` to also bubble up on the thrown `Error`?

Reply:
I dug into this more when cleaning up the CLI tool, and `response.ok` is solely based on whether the HTTP status code returned from the fetch is 200-299.  So calling it a "network" error seemed potentially inaccurate, since it could be other things.  The main change here was moving `jsonResult = (await response.json()) as XivApiResult;` above the error throw so that we actually capture the JSON output if there is an error.  Now, when you generate an XIVAPI error (say by hitting a non-existent endpoint or node), the console will actually show real info, e.g.:
```json
{
  "Error": true,
  "Subject": "XIVAPI ERROR",
  "Note": "Get on discord: https://discord.gg/MFFVHWC",
  "Message": "Game Data does not exist: Pet 85",
  "Hash": "4fb92446196ac6bba6f22a3b6964c5c20ae7ef23",
  "Ex": "Symfony\\Component\\HttpKernel\\Exception\\NotFoundHttpException",
  "ExCode": 404,
  "Url": "",
  "Debug": {
    "ID": "1UTBAx1E24010710",
    "File": "#53 xivapi.com/src/Service/Content/GameData.php",
    "Method": "GET",
    "Path": "/Pet/85",
    "Action": "App\\Controller\\XivGameContentController::contentData",
    "Code": 404,
    "Date": "2024-01-07 22:27:57",
    "Env": "prod"
  }
}
```